File: app/src/main/java/com/example/administrator/androidtest/Widget/pdfview/PdfBitmapPool.java
package com.example.administrator.androidtest.Widget.pdfview;

import android.graphics.Bitmap;

import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool;
import com.bumptech.glide.load.engine.bitmap_recycle.LruBitmapPool;
import com.example.liblog.SLog;

public class PdfBitmapPool {

    private static final String TAG = "PdfBitmapPool";
    private final BitmapPool bitmapPool;

    public PdfBitmapPool() {
        long bitmapPoolSize = 20 * 1024 * 1024; // 20MB
        bitmapPool = new LruBitmapPool(bitmapPoolSize);
    }

    public Bitmap get(int width, int height) {
        return bitmapPool.get(width, height, Bitmap.Config.ARGB_8888);
    }

    public void put(Bitmap bitmap) {
        bitmapPool.put(bitmap);
    }

    public void clear() {
        try {
            SLog.d(TAG, "clear from PdfBitmapPool: currentSize = " + ((LruBitmapPool)bitmapPool).getCurrentSize());
            bitmapPool.clearMemory();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/example/administrator/androidtest/widget/pdfview/pdfapi/pool/PdfBitmapPool.kt
package com.example.administrator.androidtest.widget.pdfview.pdfapi.pool

import android.graphics.Bitmap
import com.bear.liblog.SLog
import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool
import com.bumptech.glide.load.engine.bitmap_recycle.LruBitmapPool

class PdfBitmapPool {

    private var bitmapConfig = Bitmap.Config.RGB_565
    private val bitmapPool = LruBitmapPool(POOL_SIZE) as BitmapPool

    fun get(width: Int, height: Int): Bitmap {
        return bitmapPool[width, height, bitmapConfig]
    }

    fun put(bitmap: Bitmap?) {
        bitmapPool.put(bitmap)
    }

    fun clear() {
        try {
            bitmapPool.clearMemory()
        } catch (e: Exception) {
            SLog.d(TAG, "clear: exception message: ${e.message}")
        }
    }

    fun setBitmapConfig(config: Bitmap.Config) {
        bitmapConfig = config
    }

    companion object {
        private const val TAG = "PdfBitmapPool"
        private const val POOL_SIZE = 20 * 1024 * 1024L // 20MB
    }
}
